diff --git a/dwm.c b/dwm.c
index 8e203c0..8377717 100644
--- a/dwm.c
+++ b/dwm.c
@@ -696,7 +696,7 @@ dirtomon(int dir)
 void
 drawbar(Monitor *m)
 {
-	int x, w, tw = 0;
+	int x, w, mid, tw = 0;
 	int boxs = drw->fonts->h / 9;
 	int boxw = drw->fonts->h / 6 + 2;
 	unsigned int i, occ = 0, urg = 0;
@@ -749,8 +749,9 @@ drawbar(Monitor *m)
 
 	if ((w = m->ww - tw - x) > bh) {
 		if (m->sel) {
+			mid = (m->ww - TEXTW(m->sel->name)) / 2;
 			drw_setscheme(drw, scheme[m == selmon ? SchemeSelWin : SchemeNorm]);
-			drw_text(drw, x, 0, w - 2 * xbar, bh, lrpad / 2, m->sel->name, 0);
+			drw_text(drw, x, 0, w - 2 * xbar, bh, mid > x ? mid - x - xbar : lrpad / 2, m->sel->name, 0);
 			if (m->sel->isfloating)
 				drw_rect(drw, x + boxs, boxs, boxw, boxw, m->sel->isfixed, 0);
 		} else {
